nrf_802154_sl/README.rst

```diff
@@ -1,51 +0,0 @@
-.. _nrf_802154_sl:
```

Comment:
Another documentation file removed

Reply:
I wanted to bring those files up in subsequent PRs. If you think it is better to do it right away, then it can be done.